tests/test_utils.F90

```diff
@@ -16,40 +17,83 @@ module test_utils
     end interface
 
 contains
-    subroutine assert_equal_int(expected, actual, message)
+    subroutine assert_equal_int(expected, actual, message, status)
         integer, intent(in) :: expected, actual
         character(len=*), intent(in) :: message
+        integer, intent(out) :: status
         if (expected /= actual) then
             print *, 'FAILED: ', message
             print *, 'Expected: ', expected, ' Got: ', actual
-            error stop
+            status = ERR_ASSERT
+        else
+            status = ERR_SUCCESS
         end if
     end subroutine
 
-    subroutine assert_equal_real(expected, actual, message, tolerance)
+    subroutine assert_equal_real(expected, actual, message, tolerance, status)
         real, intent(in) :: expected, actual
         character(len=*), intent(in) :: message
         real, intent(in), optional :: tolerance
         real :: tol
+        integer, intent(out) :: status
+
         tol = 1.0e-6
         if (present(tolerance)) tol = tolerance
         if (abs(expected - actual) > tol) then
             print *, 'FAILED: ', message
             print *, 'Expected: ', expected, ' Got: ', actual
-            error stop
+            status = ERR_ASSERT
+        else
+            status = ERR_SUCCESS
```

Comment:
Not sure what's the benefit of using rc's over just stopping here. With the `error stop` + traceback you have all the info you need to identify where the failure occurred. Seems like rc's are just making the test suite more verbose.